flink-end-to-end-tests/flink-end-to-end-tests-hbase/src/test/java/org/apache/flink/tests/util/hbase/SQLClientHBaseITCase.java

```diff
@@ -0,0 +1,158 @@
+		.create(FlinkResourceSetup.builder().build());
+
+	@Rule
+	public final TemporaryFolder tmp = new TemporaryFolder();
+
+	@ClassRule
+	public static final DownloadCache DOWNLOAD_CACHE = DownloadCache.get();
+
+	private static final Path sqlToolBoxJar = TestUtils.getResourceJar(".*SqlToolbox.jar");
+	private static final Path sqlConnectorHBaseJar = TestUtils.getResourceJar(".*hbase.jar");
+	private Path shadedHadoopJar;
+	private Path sqlClientSessionConf;
+
+	public SQLClientHBaseITCase() {
+		this.hbase = HBaseResource.get();
+	}
+
+	@Before
+	public void before() throws Exception {
+		DOWNLOAD_CACHE.before();
+		Path tmpPath = tmp.getRoot().toPath();
+		LOG.info("The current temporary path: {}", tmpPath);
+		this.sqlClientSessionConf = tmpPath.resolve("sql-client-session.conf");
+
+		//download flink shaded hadoop jar for hbase e2e test using
+		shadedHadoopJar = DOWNLOAD_CACHE.getOrDownload(
+			"https://repo.maven.apache.org/maven2/org/apache/flink/flink-shaded-hadoop-2-uber/2.4.1-10.0/flink-shaded-hadoop-2-uber-2.4.1-10.0.jar", tmpPath);
+	}
+
+	@Test
+	public void testHBase() throws Exception {
+		try (ClusterController clusterController = flink.startCluster(2)) {
+			// Create table and put data
+			hbase.createTable("source_table", "a", "b");
+			hbase.createTable("sink_table", "a", "b");
+			hbase.putData("source_table", "row1", "a", "c1", "v1");
+			hbase.putData("source_table", "row1", "b", "c1", "v2");
+			hbase.putData("source_table", "row2", "a", "c1", "v3");
+			hbase.putData("source_table", "row2", "b", "c1", "v4");
+
+			// Initialize the SQL client session configuration file
+			String schemaContent = initializeSessionYaml();
+			Files.write(this.sqlClientSessionConf,
+				schemaContent.getBytes(Charsets.UTF_8),
+				StandardOpenOption.CREATE,
+				StandardOpenOption.WRITE);
+
+			// Executing SQL: read the data from HBase source_table and then write into HBase sink_table
+			insertIntoHBaseTable(clusterController);
+
+			LOG.info("Verify the sink_table result.");
+			final List<String> result = hbase.scanTable("sink_table");
+			assertThat(
+				result.toArray(new String[0]),
+				arrayContainingInAnyOrder(
+					containsString("value=v1"),
+					containsString("value=v2"),
+					containsString("value=v3"),
+					containsString("value=v4")
+				));
+			LOG.info("The HBase SQL client test run successfully.");
+		}
+
+	}
+
+	private String initializeSessionYaml() throws IOException {
+		URL url = SQLClientHBaseITCase.class.getClassLoader().getResource(HBASE_SOURCE_SINK_SCHEMA);
+		if (url == null) {
+			throw new FileNotFoundException(HBASE_SOURCE_SINK_SCHEMA);
+		}
+
+		return FileUtils.readFileUtf8(new File(url.getFile()));
+	}
+
+	private void insertIntoHBaseTable(ClusterController clusterController) throws IOException {
+		LOG.info("Executing SQL: HBase source_table -> HBase sink_table");
+		String sqlStatement = "INSERT INTO MyHBaseSink\n" +
+			"	SELECT rowkey, a, b from MyHBaseSource";
```

Comment:
Could you update this to a more complex query? e.g. 

```sql
SELECT rowkey, a, ROW(RegReplace(a.c1, 'v', 'value')) from MyHBaseSource";
```

It uses udf and ROW to construct column family. We need to register the UDF RegReplace then. 


```yaml
functions:
  - name: RegReplace
    from: class
    class: org.apache.flink.table.toolbox.StringRegexReplaceFunction
```